### PR TITLE
syncer: renaming and some little refactor

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 

- rename "current" position to "next" position, want to express "current event's end position is next position".
- extract some logic into methods

### What is changed and how it works?
⬆️

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

Related changes

### CODE REVIEW

File: syncer/syncer.go [hunk 1]
```diff
@@ -1826,103 +1754,14 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
...
-		annotate = "try to sync table in shard group"
-		// meets DDL that will not be processed in sequence sharding
-		if !active {
-			s.tctx.L().Info("skip in-activeDDL", zap.String("event", "query"), zap.String("source", source), zap.Strings("ddls", needHandleDDLs), zap.ByteString("raw statement", ev.Query), zap.Bool("in-sharding", needShardingHandle), zap.Stringer("start location", startLocation), zap.Bool("is-synced", synced), zap.Int("unsynced", remain))
-			return nil
-		}
-	}
-
-	s.tctx.L().Info(annotate, zap.String("event", "query"), zap.String("source", source), zap.Strings("ddls", needHandleDDLs), zap.ByteString("raw statement", ev.Query), zap.Bool("in-sharding", needShardingHandle), zap.Stringer("start location", startLocation), zap.Bool("is-synced", synced), zap.Int("unsynced", remain))
+	s.tctx.L().Info("start to handle ddls in normal mode", zap.String("event", "query"), zap.Strings("ddls", needHandleDDLs), zap.ByteString("raw statement", ev.Query), log.WrapStringerField("location", ec.currentLocation))
```
Comment (lance6716) [inline syncer/syncer.go:1765]: this is the original logic of `if s.cfg.ShardMode == "" {}`, git diff doesn't recognize it

File: syncer/syncer.go [hunk 2]
```diff
@@ -1828,103 +1757,14 @@ func (s *Syncer) handleQueryEvent(ev *replication.QueryEvent, ec eventContext) e
...
-		return nil
-	}
-
-	// handle shard DDL in optimistic mode.
+	// handle shard DDL
 	if s.cfg.ShardMode == config.ShardOptimistic {
 		return s.handleQueryEventOptimistic(ev, ec, needHandleDDLs, needTrackDDLs, onlineDDLTableNames)
+	} else if s.cfg.ShardMode == config.ShardPessimistic {
+		return s.handleQueryEventPessimistic(ev, ec, needHandleDDLs, needTrackDDLs, onlineDDLTableNames, ddlInfo, sqls)
 	}
```
Comment (GMHDBJD) [inline syncer/syncer.go:1765]: Also refine the normal mode？😂

Comment (lance6716) [reply syncer/syncer.go:1767]: done in https://github.com/pingcap/dm/pull/896/commits/575352625260a2b8ae0e489d24c4b09a37dd407a

**Pull request closed without merging**